pkg/operatorsource/catalogsourceconfigcrud.go

```diff
@@ -0,0 +1,72 @@
+package operatorsource
+
+import (
+	"context"
+
+	v1 "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CatalogSourceConfigCRUD defines CRUD operations for CatalogSourceConfig object in the cluster
+type CatalogSourceConfigCRUD interface {
+	CreateCatalogSourceConfig(ctx context.Context, in *v1.OperatorSource, manifests string) error
+	GetCatalogSourceConfig(ctx context.Context, in *v1.OperatorSource) (cscExisting v1.CatalogSourceConfig, err error)
+	UpdateCatalogSourceConfig(ctx context.Context, in *v1.OperatorSource, cscExisting v1.CatalogSourceConfig, manifests string) error
+}
+
+// cscCrud is an implementation of the ConfiguringReconcilerCRUD interface
+type cscCrud struct {
+	client client.Client
+}
+
+// NewCatalogSourceConfigCRUD returns a cscCrud that can perform
+// CRUD operations on a Catalogsourceconfig object
+func NewCatalogSourceConfigCRUD(client client.Client) CatalogSourceConfigCRUD {
+	return &cscCrud{
+		client: client,
+	}
+}
+
+// CreateCatalogSourceConfig creates a new CatalogSourceConfig
+func (h *cscCrud) CreateCatalogSourceConfig(ctx context.Context, in *v1.OperatorSource, manifests string) error {
```

Comment:
The `in` should be `owner` and it technically can be any object. This comment applies everywhere you are doing this.

Reply:
+1